src/crnlib.rs

```diff
@@ -0,0 +1,144 @@
+use crate::crunch::crn_decomp::CrnHeader;
+
+// Supported compressed pixel formats.
+// Basically all the standard DX9 formats, with some swizzled DXT5 formats
+// (most of them supported by ATI's Compressonator), along with some ATI/X360 GPU specific formats.
+#[derive(PartialEq, PartialOrd)]
+#[repr(C)]
+pub enum CrnFormat {
+    CCrnfmtInvalid = -1,

```

Comment:
drop `CCrnfmt` prefix from enum members. The reason it's like that in C is because enum members get added to global namespace. In Rust they are in the per-enum namespace, so adding prefixes like this is actually bad as it leads to duplication (`CrnFormat::CCrnfmtInvalid` vs `CrnFormat::Invalid`}

Reply:
In fact, many codebases would have this called `Format` and refer to it as `crn::Format::Invalid`